proposals/0000-severe-warnings.rst

```diff
@@ -0,0 +1,157 @@
+  
+  It behaves like the other groups in that ``-Wno=severe``, ``-Wwarn=severe`` and ``-Werror=severe`` behave as if the flag
+  had been specified for each warning in the groups.
+  
+* The warning group is initialized with the following warning categories:
+
+  * ``missing-methods``
+  * ``missing-fields``
+
+*  The set of allowed names of custom warning categories (`#541 <https://github.com/ghc-proposals/ghc-proposals/pull/541>`_) is
+   extended by those prefixed with ``xs-*`` (for “custom extended – severe”). The flags
+   ``-Wno-severe``, ``-Wwarn=severe`` and ``-Werror=severe``
+   also affect all custom warnings with a name starting in ``xs-*``.
+ 
+
+Examples
+--------
+TODO
+
+
+Effect and Interactions
+-----------------------
+TODO
+
+
+Costs and Drawbacks
+-------------------
+In terms of GHC development, this is a modest extension of the existing warning category and group infrastructure.
+
+In terms of breakage, @phadej has made an impact analysis at <https://github.com/ghc-proposals/ghc-proposals/issues/544#issue-1410125536>.
+
+
+Alternatives
+------------
+
+* Naming the group.
+
+  I suggest to use the name ``severe``, which does not have ``error`` in it (e.g. ``errors-by-default``), because else
+  ``-Werror=errors-by-default`` or something looks kinda strange, and after someone says ``-Wwarn=severe``, the ``severe`` group still
+  exists, but it not an error.
+  
+  I briefly considered ``-Wfatal``, but that’s a lie – these errors are *not* ``fatal``, else we couldn’t turn them
+  into warnings.
+  
+* We could add more warnings to the group right away.
+
+* In particular, we could make ``partial-type-signatures`` a normal warning, in ``-Wsevere`` by default, and ``-XPartialTypeSignatures``
+  becomes a synonym for ``-Wwarn=partial-type-signatures``.
+  
+  (Or should ``-XPartialTypeSignatures`` be a flag that changes the *warning group* of ``partial-type-signatures``? Not sure).
+
+* An additional motivation is the currently circulating idea that some features that are language *extensions* right now (``-XFoo``) can
+  simply become part of the default “normal” language, together with a new warning category (``foo``) that’s off by default normally, and 
+  the extension flags ``-XNoFoo`` or ``-XHaskell2010`` simply have the same effect as ``-Werror=foo``.
+  
+  Although now that I write it out, it seems that we don't actually need ``-Wsevere`` for that. Maybe it’s useful to *strongly deprecate* features,
+  by introducing a warning about their use, and raising its severity from ``-Wcompat`` to ``-Wdefault`` to ``-Werror`` over time.
+
+* We could guard this change behind a suitable language extension, so that ``Haskell2010`` stays untouched. It could be the default eventually, but
+  would not affect code under ``Haskell2010`` or ``GHC2021``.
+  
+  It would set precedent for langauge extensions changing the default mode (warning vs. errors) of warnings, and I’d propose that the semantics
```

Comment:
```suggestion
  It would set precedent for language extensions changing the default mode (warning vs. errors) of warnings, and I’d propose that the semantics
```